shared-module/synthio/Note.c

```diff
@@ -135,6 +135,24 @@ void common_hal_synthio_note_set_ring_waveform(synthio_note_obj_t *self, mp_obj_
     self->ring_waveform_obj = ring_waveform_in;
 }
 
+mp_int_t common_hal_synthio_note_get_loop_start(synthio_note_obj_t *self) {
+    return self->loop_start;
+}
+
+void common_hal_synthio_note_set_loop_start(synthio_note_obj_t *self, mp_int_t value_in) {
+    mp_int_t val = mp_arg_validate_int_range(value_in, 0, 32767, MP_QSTR_loop_start);
```

Comment:
I used 16384 as the maximum length of a waveform (which maybe should have been 16383 since the upper bound is inclusive); this should probably match, which means it would be good if there was a `#define` for it in `shared-module/synthio/__init__.h`. 

I don't have any notes that show why I used 16384 instead of a higher value like 32767. I may have been concerned about overflow in arithmetic.

Reply:
Allowing a higher maximum size would definitely be a plus for sampling, but I vote we leave that for a separate PR. Creating the `#define` will help make the process of changing that value in the future easier, though.